source/common/profiler/profiler.h

```diff
@@ -38,6 +38,24 @@ class Cpu {
  * Process wide heap profiling
  */
 class Heap {
+public:
+  /**
+   * @return whether the profiler is enabled or not.
+   */
+  static bool profilerEnabled();
+
+  /**
```

Comment:
I think we probably want a separate method indicating whether the current build supports profiling.

Reply:
Counterpoint to my suggestion here: that might result in lines-of-code that are hard to cover in the coverage run. Maybe you should propagate the ifdef instead, so that coverage doesn't punish you for lines you can't hit.

Reply:
Add separate method. It is required to implement actionable error message. We have to sacrify coverage...

Reply:
cool. will do

Reply:
I looked into the existing usage of 
`ENVOY_CONFIG_COVERAGE`, the existing case is debugPrint which doesn't involve branch logic. 
https://github.com/envoyproxy/envoy/search?q=ENVOY_CONFIG_COVERAGE&unscoped_q=ENVOY_CONFIG_COVERAGE

I will update the commit and comment the convolution logic if we introduce ENVOY_CONFIG_COVERAGE. Maybe you are referring different approach of using `ENVOY_CONFIG_COVERAGE`